app/controllers/campaigns_controller.rb

```diff
@@ -7,12 +7,13 @@ def display_skip_to_main_navigation?
     false
   end
 
+  ::CampaignPage::Campaign::WHITELIST.each do |campaign_page|
+    method_name = campaign_page.underscore
+    class_eval "def #{method_name}; end"
```

Comment:
Can these methods be private then? And other thing, I would use define_method instead of class_eval. More details here: http://tenderlovemaking.com/2013/03/03/dynamic_method_definitions.html

In case you've doubts about to use define_method:

```ruby
2.1.2 :001 > FOO = %w(a b c)
 => ["a", "b", "c"]
2.1.2 :002 > FOO.each do |foo|
2.1.2 :003 >     end
 => ["a", "b", "c"]
2.1.2 :004 > FOO.each do |foo|
2.1.2 :005 >     define_method "#{foo}" do
2.1.2 :006 >       puts "#{foo}"
2.1.2 :007?>     end
2.1.2 :008?>   end
 => ["a", "b", "c"]
2.1.2 :009 > a
a
 => nil
2.1.2 :010 > b
b
 => nil
2.1.2 :011 > c
c
 => nil
```